### PR TITLE
Nikon alignment shift correction

### PR DESCRIPTION
**Pull request by amuskens**
Almost ready.

### CODE REVIEW

File: src/odemis/model/_metadata.py [hunk 1]
```diff
@@ -134,6 +134,9 @@
 MD_HFW_SLOPE = "HFW slope"  # (float, float) HFW related SEM image shift, slope of linear fit
 MD_SPOT_SHIFT = "Spot shift"  # (float, float), SEM spot shift in percentage of HFW
 
+# The following metadata is for correction on the Nikon C
+MD_SHIFT_LOOKUP = "Pixel shift compensation LUP"
```
Comment (pieleric) [inline src/odemis/model/_metadata.py]: I'd name it "Pixel shift compensation table"
Also add info on the type in a comment, eg `# dict (int (resolution X) -> dict (float (dwell time) -> tuple of 4 floats (correction factors)))

File: scripts/nikon_shift_correction.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env python
```
Comment (pieleric) [inline scripts/nikon_shift_correction.py]: It's not clear what this script is for. It feels like it's a mix between analyse_shifts.py and curvefit_test.py.

File: scripts/curvefit_test.py [hunk 3]
```diff
@@ -0,0 +1,227 @@
+import sys
```
Comment (pieleric) [inline scripts/curvefit_test.py]: curvefit_test.py and make_shift_calib.py have essentially the same issues:
* Make them executable (ie, put `#!/usr/bin/env python2` at the top
* Put a short comment explaining what the script is for and what it does. Finish with an example command line to show how to call the script
* This scripts are very specific to the nikonc. So either modify the script names to highlight they are not (for example) general curve-fitting, but very specific one for confocal mirror movements. Or you could just move the scripts to the nikonc driver repo (and still rename to `curvefit_mirror.py` or something of this style).

**Comment by amuskens** [comment]
Made some changes to address comments.

**Comment by amuskens** [comment]
Added some changes - which are tested a bit. With the latest configuration, I get this:

WARNING:root:Bad value @ res 1024, z 50.000000, dt 0.000001 s. Actual shift was 0.003906, calculated was -13.050443
WARNING:root:Bad value @ res 128, z 1.000000, dt 0.000018 s. Actual shift was -21.800000, calculated was -34.193536
WARNING:root:Bad value @ res 128, z 1.000000, dt 0.000015 s. Actual shift was -45.100000, calculated was -32.278727
WARNING:root:Bad value @ res 64, z 1.250000, dt 0.000031 s. Actual shift was -91.500000, calculated was -73.888497
WARNING:root:Bad value @ res 64, z 1.250000, dt 0.000015 s. Actual shift was -95.800000, calculated was -83.771304
WARNING:root:Bad value @ res 64, z 1.000000, dt 0.000018 s. Actual shift was -65.100000, calculated was -83.678272
WARNING:root:Bad value @ res 64, z 1.000000, dt 0.000031 s. Actual shift was -107.800000, calculated was -84.179147
WARNING:root:Bad value @ res 64, z 1.000000, dt 0.000015 s. Actual shift was -99.300000, calculated was -82.274516
WARNING:root:Bad value @ res 64, z 1.000000, dt 0.000024 s. Actual shift was -70.700000, calculated was -85.215662
WARNING:root:Bad value @ res 64, z 4.000000, dt 0.000019 s. Actual shift was -49.200000, calculated was -70.513513
WARNING:root:Bad value @ res 64, z 4.000000, dt 0.000038 s. Actual shift was -33.300000, calculated was -49.668887
WARNING:root:Bad value @ res 64, z 4.000000, dt 0.000031 s. Actual shift was -41.600000, calculated was -57.971845
WARNING:root:Bad value @ res 64, z 4.000000, dt 0.000015 s. Actual shift was -60.700000, calculated was -74.752213
WARNING:root:Bad value @ res 64, z 8.000000, dt 0.000019 s. Actual shift was -24.100000, calculated was -51.908137
WARNING:root:Bad value @ res 64, z 8.000000, dt 0.000038 s. Actual shift was -15.400000, calculated was -36.074864
WARNING:root:Bad value @ res 64, z 8.000000, dt 0.000031 s. Actual shift was -19.700000, calculated was -42.351796
WARNING:root:Bad value @ res 64, z 8.000000, dt 0.000015 s. Actual shift was -29.700000, calculated was -55.187537
WARNING:root:Bad value @ res 64, z 1.200000, dt 0.000047 s. Actual shift was -66.500000, calculated was -52.414303
WARNING:root:Bad value @ res 64, z 1.500000, dt 0.000031 s. Actual shift was -83.300000, calculated was -69.727905
WARNING:root:Bad value @ res 64, z 1.500000, dt 0.000015 s. Actual shift was -96.900000, calculated was -85.325929
WARNING:root:Bad value @ res 256, z 1.000000, dt 0.000006 s. Actual shift was 1.900000, calculated was -8.515879
WARNING:root:2.796272 % (21 / 751) of tested points were outside of the pixel shift threshold.

**Comment by amuskens** [comment]
Rebased. Should be able to merge?

File: scripts/nikonc2_curvefit_test.py [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+# -*- coding: utf-8 -*-
+'''
+Created on 15 October 2018
+
+@author: Anders Muskens
+
+Testing script for determining the curve fit for analyze-shifts.py TSV files.
+
+'''
+
```
[the remark this answers is not in the document]
Comment (amuskens) [reply scripts/nikonc2_curvefit_test.py]: Should be fixed now.

File: src/odemis/model/_metadata.py [hunk 5]
```diff
@@ -134,6 +134,10 @@
 MD_HFW_SLOPE = "HFW slope"  # (float, float) HFW related SEM image shift, slope of linear fit
 MD_SPOT_SHIFT = "Spot shift"  # (float, float), SEM spot shift in percentage of HFW
 
+# The following metadata is for correction on the Nikon Confocal
```
Comment (pieleric) [inline src/odemis/model/_metadata.py]: Please drop your changes to this files, as they are already in master.

**Pull request merged**